core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -865,25 +865,41 @@ public void deleteInode(RpcContext rpcContext, LockedInodePath inodePath, long o
     }
   }
 
+  private boolean checkPinningValidity(Set<String> pinnedLocations) {
+    List<String> mediaList = ServerConfiguration.getList(
+        PropertyKey.MASTER_TIERED_STORE_GLOBAL_MEDIA, ",");
+    for (String medium : pinnedLocations) {
+      if (!mediaList.remove(medium)) {
+        // mediaList does not contains medium
+        return false;
+      }
+    }
+    return true;
```

Comment:
Since `pinnedLocations` is a set, you can check it without using the `remove` call.